File: Convert_Types/Decimal_Hexadecimal.c
#include <stdio.h>
void main()
{
    int decimal; 

    printf("Enter a decimal number: ");
    scanf("%d", &decimal);

    char hex[20];
    int i = 0;

    while (decimal > 0)
    {
        int rem = decimal % 16;

        if (rem < 10)
            hex[i] = rem + '0';
        else
            hex[i] = (char)(rem + 55); // A,B,C,D,E,F

        decimal /= 16;
        i++;
    }

    printf("Hexadecimal Equivalent = ");

    for (i = i - 1; i >= 0; i--)
        printf("%c", hex[i]);
}

File: Convert_Types/Decimal_Hexadecimal.java
import java.util.Scanner;

public class Decimal_Hexadecimal {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);

        System.out.println("Enter a decimal number: ");
        int decimal = sc.nextInt();

        String hex = "";

        while (decimal > 0) {
            int rem = decimal % 16;

            if (rem < 10)
                hex = rem + hex;
            else
                hex = (char)(rem + 55) + hex;//A,B,C,D,E,F

            decimal /= 16;
        }

        System.out.println("Hexadecimal Equivalent = " + hex);
    }
}
